### ISSUE TITLE
Parameters for LAMP.ActivitySpec.Create should be changed (LAMP Server)

### ISSUE THREAD

**Issue by lukeoftheshire** [BIDMCDigitalPsychiatry/LAMP-platform#391]
Currently, the ActivitySpec _insert method is not implemented in an ideal way. Currently, the object used as a parameter must have the key 'name', when it should probably be 'id'.

The below calls were done in Python.
For example (on the staging server), calling `LAMP.ActivitySpec.create({'name':'org.digitalpsych.luke.activity_example'})` both works and creates the ActivitySpec
```
{'id': 'org.digitalpsych.luke.activity_example',
   'name': 'org.digitalpsych.luke.activity_example',
   'help_contents': None,
   'script_contents': None,
   'static_data_schema': {},
   'temporal_slice_schema': {},
   'settings_schema': {}}
```
   
   Calling `LAMP.ActivitySpec.create({'id':'org.digitalpsych.luke.activity_example_2'})` also works, in that no errors are thrown, but the added ActivitySpec is:
  ``` 
  {'id': '615359a1462a02249cedf7ea',
   'name': '615359a1462a02249cedf7ea',
   'help_contents': None,
   'script_contents': None,
   'static_data_schema': {},
   'temporal_slice_schema': {},
   'settings_schema': {}}
```

Thus, when given an object with an id parameter, the current _insert function in LAMP-server does not create an ActivitySpec in the DB with the appropriate id - instead, it creates an ActivitySpec with both a name and id are (apparently) an identical randomly assigned id.
   
In an ideal implementation, the following changes should happen:
1.  There should not key-value pairs with identical values in the database object - this should probably be resolved by simply not having a 'name' parameter and instead only using 'id' - 'id' is more consistent with the rest of LAMP.
2. Inserting an activityspec into the database should be able to use the 'id' key of a provided object (at the least in *addition* to using the 'name' key, but probably *instead of* the name key )

In other words, an improved implementation could produce a result like the one from the first function call above, but by calling the equivalent of the second function call above.

**Comment by avaidyam** [comment]
@Linoy339 Could you remove the `name` parameter from `_select` functions and switch to using `id` instead of `name` for both ActivitySpec and SensorSpec for Mongo and Couch adapters? It should be a quick rename.

**Comment by Linoy339** [comment]
> @Linoy339 Could you remove the `name` parameter from `_select` functions and switch to using `id` instead of `name` for both ActivitySpec and SensorSpec for Mongo and Couch adapters? It should be a quick rename.

Yes. We can do it. 
Just to confirm: Is that the only change needed?

**Comment by avaidyam** [comment]
Yes, for now.
